projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/upload/JobUploadFilterListValidator.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2021-2023 VMware, Inc.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.vmware.taurus.service.upload;
+
+import java.nio.file.Path;
+import java.util.Arrays;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.io.FilenameUtils;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+/**
+ * This class filters out the job directory from forbidden files so that it can be uploaded without
+ * them. Forbidden files are specified in a comma separated list in apache tika format. Full list of
+ * supported files can be found - https://tika.apache.org ; This operation is intended to allow the
+ * upload of a data job by deleting any files specified in the:
+ * upload.validation.fileTypes.filterlist property.
+ */
+@Service
+@Slf4j
+public class JobUploadFilterListValidator extends AbstractJobFileValidator {
+
+  private final String[] filterList;
+
+  public JobUploadFilterListValidator(
+      @Value("${upload.validation.fileTypes.filterlist:}") String[] filterList) {
+    this.filterList = filterList;
+  }
+
+  @Override
+  String[] getValidationTypes() {
+    return this.filterList;
+  }
+
+  @Override
+  String detectFileType(Path filePath) {
+    return FilenameUtils.getExtension(filePath.getFileName().toString());
```

Comment:
detectFileType should be the same for both filtering and allowing 

Otehrwise filterList and allowedList would have different format which would be very error prone. 
Also likely we may want to allow files by extension and filter out files by detected tika type. 

I would suggest to have 2 methods 

detectFileType -> which would return the Tika type 
detectFileExtesion -> which would return the extension. 
Both would be in the parent class. 

then `matchTypes` method would accept (detectedType, detectedExtension). 

We would need 2 properties allowedTypes and allowedExtensions and filterTypes and filterExtensions

as we should be explicit about the format of the values and we cannot do that in one property.